### PR TITLE
Adding identity provider to the user collection and some refactoring …

### PR DESCRIPTION
**Pull request by DanielCoello**
**Description:**
Added the identity_provider to the user collection, so we can know from which identity is the user coming. Also, did some refactoring to the local (so it only searches the user if the identity_provider is null) and salesforce (changed the env variables so we only have the api urls). 

@pencilblue/owners

### CODE REVIEW

File: include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js [hunk 1]
```diff
@@ -24,9 +25,9 @@ module.exports = (pb) => {
         pluginService = Promise.promisifyAll(pluginService);
         const settings = await pluginService.getSettingsKVAsync('tn_join');
```
Comment (lgenet) [inline include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js:26]: Remember, we need to change this from tn_join before this goes live.

File: include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js [hunk 2]
```diff
@@ -2,20 +2,21 @@ const CustomStrategy = require('passport-custom').Strategy;
 const strategyServices = require('./strategy_services');
 const request = require('request-promise');
 const Promise = require('bluebird');
-let SALESFORCE_OAUTH_TOKEN_URL, SALESFORCE_OAUTH_AUTHORIZE_URL, state, SALESFORCE_API_URL;
+let state, salesforceAPIUrl;
 const isSandbox = process.env.USE_SALESFORCE_SANDBOX === 'true';
+const OAUTH_TOKEN_SERVICE = '/services/oauth2/token';
```
Comment (lgenet) [inline include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js:7]: Good Change, if these are not 
A.) Secret
B.) Environment Dependent
Then there is no reason to have them in the .evn files.

File: include/koa/authentication/passport-strategies/strategy_services.js [hunk 3]
```diff
@@ -50,10 +51,10 @@ function _getQuery(loginContext, pb, ignorePassword) {
     return query;
 };
 
-async function getUser(loginContext, done, pb, ignorePassword = false) {
+async function getUser(loginContext, identityProvider, done, pb, ignorePassword = false) {
```
Comment (lgenet) [inline include/koa/authentication/passport-strategies/strategy_services.js:54]: There was once upon a time a convention for passing in params.  Like PB had a certain place and callback/done was last.  I dont remember nor do I care particularly at this point.

**Review by lgenet — APPROVED** [review]

File: include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js (hunk 1 above)
Comment (DanielCoello) [reply include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js:26]: Yes, I've already created the card for this. Going to do it when we deploy cms-koa branch to master

File: include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js [hunk 4]
```diff
@@ -76,7 +78,7 @@ module.exports = (pb) => {
         try {
             const settings = await getSalesforceSettings(req);
             const options = {
-                url: `${SALESFORCE_OAUTH_AUTHORIZE_URL}?client_id=${settings.salesforce_client_id}&redirect_uri=https://${req.header.host}/login/salesforce/callback&response_type=code&state=${state}`,
+                url: `${salesforceOAUTHAuthorizeService}?client_id=${settings.salesforce_client_id}&redirect_uri=https://${req.header.host}/login/salesforce/callback&response_type=code&state=${state}`,
                 method: 'POST'
             };
             return done(null, options);
```
Comment (DanielCoello) [inline include/koa/authentication/passport-strategies/salesforce_authentication_strategy.js:84]: @JoseDomingoMoraNunezCB

**Pull request merged**